datafusion/core/tests/sql/joins.rs

```diff
@@ -231,3 +231,32 @@ async fn join_change_in_planner_without_sort_not_allowed() -> Result<()> {
     }
     Ok(())
 }
+
+#[tokio::test]
+async fn test_smj_with_join_filter_fitering_all() -> Result<()> {
+    let ctx: SessionContext = SessionContext::new();
+
+    let sql = "set datafusion.optimizer.prefer_hash_join = false;";
+    let _ = ctx.sql(sql).await?.collect().await?;
+
+    let sql = "set datafusion.execution.batch_size = 1";
+    let _ = ctx.sql(sql).await?.collect().await?;
+
+    let sql = "
```

Comment:
is there any reason this isn't a slt test (in https://github.com/apache/datafusion/blob/main/datafusion/sqllogictest/test_files/sort_merge_join.slt)? Looks like it would work just fine

Reply:
No, I simply copy the reproducer from the issue. Let me move it to sort_merge_join.slt.